rp235x-hal/src/gpio/func.rs

```diff
@@ -48,6 +48,8 @@ pub enum DynFunction {
     /// The 'Clock' function. This can be used to input or output clock references to or from the
     /// rp235x.
     Clock,
+    /// The XIP CS1 function.
+    XipCs1,
```

Comment:
Maybe this is the same DynFunction as Clock? They have the same FUNCSEL value.

Reply:
I am not sure I understand your comment.
Are you proposing instead of having a DynFunction for XipCs1 it reuses Clock as they have same FUNCSEL value?

Would that mean also changing/removing? https://github.com/rp-rs/rp-hal/pull/873/files#diff-514435c8bc1dac690be86445ca032b00cea05a617a1fef193197402cf99d94e0R211:
```rust
pin_valid_func!(bank0 as Gpio, [XipCs1], [0, 8, 19, 47]);
```

I am not sure this would be a correct change as although both Clock and XipCs1 use the same FUNCSEL value; the pins that can be Clock and the pins that can be XipCs1 are not the same.

If the plan is not to change `rp235x-pac` I think that we keep both `DynFunction::Clock` and `DynFunction::XipCs1` and simply point them to the same `FUNCSEL_A::GPCK` as currently done on: https://github.com/rp-rs/rp-hal/pull/873/files#diff-e4e5afea6f39b9d83036b039760824fc1f6ad0138f1106bdaa18d11d4b932577R171

This way the api exposed by the HAL is sound and matches expected behavior and the discrepancy inside  `rp235x-pac` 's `FUNCSEL_A` is not exposed.

Unless I am misunderstanding your comment.

Reply:
> Are you proposing instead of having a DynFunction for XipCs1 it reuses Clock as they have same FUNCSEL value?

I was. But I think you're right - it's probably better to keep them separate.